src/scenes/Event/Leaderboard/index.js

```diff
@@ -28,42 +33,64 @@ const tabs = [messages.mostNBOT, messages.biggestWinner];
 @observer
 export default class Leaderboard extends React.Component {
   handleNext = () => {
-    const { ui: { location } } = this.props.store;
-    if (location === Routes.LEADERBOARD) {
-      const { leaderboard } = this.props.store;
-      leaderboard.activeStep += 1;
-    } else {
-      const { eventPage } = this.props.store;
-      eventPage.activeStep += 1;
-    }
+    const { leaderboard } = this.props.store;
+    leaderboard.activeStep += 1;
   };
 
   handleBack = () => {
-    const { ui: { location } } = this.props.store;
-    if (location === Routes.LEADERBOARD) {
-      const { leaderboard } = this.props.store;
-      leaderboard.activeStep -= 1;
-    } else {
-      const { eventPage } = this.props.store;
-      eventPage.activeStep -= 1;
-    }
+    const { leaderboard } = this.props.store;
+    leaderboard.activeStep -= 1;
   };
 
+  renderHeader = () => {
+    const { classes } = this.props;
+    return (
+      <Grid container className={classes.grid} justify="flex-start">
+        <Grid item xs={2} sm={2}>
+          <FormattedMessage id="leaderboard.ranking" defaultMessage="Ranking" />
+        </Grid>
+        <Grid item xs={8} sm={8}>
+          <FormattedMessage id="leaderboard.address" defaultMessage="Address" />
+        </Grid>
+        <Grid item xs={2} sm={2}>
+          <FormattedMessage id="leaderboard.amount" defaultMessage="Amount" />
+        </Grid>
+      </Grid>
+    );
+  }
+
+  renderEntry = (row, index) => {
+    const { classes, intl, store: { naka: { account } } } = this.props;
+    const { betterAddress, amount } = row;
+    if (!betterAddress) return;
+    const ranking = (index <= 2 && <img src={`/images/ic_${index + 1}_cup.svg`} alt='cup' />)
+                    || (index === 3 && '👍') || (index >= 4 && '✊');
+
+    const address = (account && account.toLowerCase() === betterAddress && intl.formatMessage(messages.strYou)) || `${betterAddress.slice(0, 6)}...${betterAddress.slice(-6)}`;
```

Comment:
suggest to make a reusable component `WalletAddress` with props to give the option to ellipsize it or not